catalog/video/video-mpegts-stream/src/main/java/org/codice/alliance/video/stream/mpegts/netty/UdpStreamProcessor.java

```diff
@@ -316,18 +338,40 @@ public PacketBuffer getPacketBuffer() {
      * of IDR boundaries.
      */
     public void shutdown() {
-        try {
-            LOGGER.trace("Shutting down stream processor.");
-            packetBuffer.cancelTimer();
-            streamShutdownPlugin.onShutdown(context);
-        } catch (StreamShutdownException e) {
-            LOGGER.debug("unable to shutdown", e);
-        }
+        LOGGER.trace("Shutting down stream processor.");
+        packetBuffer.cancelTimer();
+
+        Security.runAsAdmin(() -> {
```

Comment:
You need to elevate to the Java system admin in order to retrieve the system subject. This makes it more obvious in code that you're elevating permissions. We probably shouldn't hang on to that subject, however. It looks like it is an instance field? We should just use it and discard it. It is already cached within the getSystemSubject() call, so it doesn't save anything performance wise. It does, however, make things a little more dangerous if we can then continue using that system subject in other areas outside of this block.